spring-integration-core/src/main/java/org/springframework/integration/router/AbstractMessageRouter.java

```diff
@@ -171,24 +173,20 @@ protected void handleMessageInternal(Message<?> message) {
 			int sequenceSize = results.size();
 			int sequenceNumber = 1;
 			for (MessageChannel channel : results) {
-				final Message<?> messageToSend =
-						!this.applySequence ? message : (this.getMessageBuilderFactory()
+				final Message<?> messageToSend;
+				if (!this.applySequence) {
+					messageToSend = message;
+				}
+				else {
+					UUID id = message.getHeaders().getId();
+					messageToSend = getMessageBuilderFactory()
 								.fromMessage(message)
-								.pushSequenceDetails(message.getHeaders().getId(), sequenceNumber++, sequenceSize)
-								.build());
+								.pushSequenceDetails(id == null ? IntegrationObjectSupport.generateId() : id,
```

Comment:
The `AbstractMessageRouter.` is an `IntegrationObjectSupport` already: no reason to use explicit method. you can just omit the class. Or Sonar won't like it still?